addons/website_sale/controllers/main.py

```diff
@@ -692,6 +692,43 @@ def extra_info(self, **post):
     # Payment
     # ------------------------------------------------------
 
+    def _get_shop_payment_values(self, order, **kwargs):
+        if order and order.partner_shipping_id.id:
+            shipping_partner_id = order.partner_shipping_id.id
+        elif order:
+            shipping_partner_id = order.partner_invoice_id.id
+        else:
+            shipping_partner_id = False
```

Comment:
why not use great properties of great new API and just write `shipping_partner_id = order.partner_shipping_id or order.partner_invoice_id`?